eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Certs.hs

```diff
@@ -254,3 +254,28 @@ instance
   where
   wrapFailed = CertFailure
   wrapEvent = CertEvent
+
+-- | Update dormant expiry for all DReps that are active.
+-- And also reset the `numDormantEpochs` counter.
+updateDormantDRepExpiry ::
+  -- | Current Epoch
+  EpochNo ->
+  -- | Number of dormant epochs
+  EpochNo ->
+  VState era ->
+  VState era
+updateDormantDRepExpiry currentEpoch numDormantEpochs vState =
+  if numDormantEpochs == EpochNo 0
+    then vState
+    else
+      vState
+        & vsNumDormantEpochsL .~ EpochNo 0
+        & vsDRepsL %~ fmap updateExpiry
+  where
+    updateExpiry =
+      drepExpiryL
+        %~ \currentExpiry ->
+          let actualExpiry = binOpEpochNo (+) numDormantEpochs currentExpiry
+           in if actualExpiry < currentEpoch
```

Comment:
Just for my benefit, the difference from the previous implementation for the case in which we have proposals  and non-zero dormant epochs is here, right?  We don't extend the expiry if it's already expired, am I understanding correctly?

Reply:
Yes, you are correct. :+1: